tests/tasks/setup_mock_wifi.yml

```diff
@@ -0,0 +1,91 @@
+# SPDX-License-Identifier: BSD-3-Clause
+---
+- name: Install EPEL on enterprise Linux for hostapd
+  # yamllint disable-line rule:line-length
+  command: yum install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-{{ ansible_distribution_major_version }}.noarch.rpm
+  args:
+    warn: false
+    creates: /etc/yum.repos.d/epel.repo
+  when:
+    - ansible_distribution in ['RedHat', 'CentOS']
+
+- name: Install packages required to set up mock wifi network
+  package:
+    name:
+      - hostapd
+      - NetworkManager
```

Comment:
Installing NM and wpa_supplicant should probably be left to the role to test its installation checks.

Reply:
I was basically just copying what was in the NM integration tests but I'm pretty sure it will work without this

Reply:
Tested removing both of these but it appears hostapd requires wpa_supplicant to be running

Reply:
How about we split the test into two phases:

1) Setup a wireless profile without the hostapd maybe with state present/down and check that the profile is generated correctly
2) Setup hostapd etc but without touching wpa_supplicant or restarting NM (this should now work because the first step should already have done this
3) Change the profile to state:up to do the actual integration test

What do you think? I feel a little bit uneasy that the test replicates some steps that are actually part of the role logic. This is also the issue with the other steps here that (re)start NM.

Reply:
@tyll I tried implementing this but it appears the role will try and "down" the profile even if it is already down (step 1). This fails for a wireless connection unless I commented out most of the "run_action_down" function. I could add "failed_when: false" to the initial connection creation but this is probably not a great solution

I can try working on this some more but maybe it would be better to just confirm all the variables are set correctly?
i.e. checking __network_wpa_supplicant_required == true, __network_wireless_connections_defined == true or something?

On a default RHEL7 installation I believe NetworkManager and wpa_supplicant will already be installed and running so I don't think it is a huge issue if they are restarted before we run the role.